### DISCUSSION TITLE
@mantine/forms `onBlur` callback/subscription

### DISCUSSION THREAD

**Discussion by imsherrill** [mantinedev/mantine#6185] [Feature requests]
I'm thinking about scalable ways to implement a submit-on-blur paradigm using Mantine. I know that I could do something like the following:
```
<TextInput
 {...form.getInputProps('field')}
  onBlur={(e) => {
    form.getInputProps('field').onBlur(e)
	form.onSubmit(() => {})()
  }}
/>
```

but its not particularly scalable since that'd need to be on every input within the form. I've also considering using `useEffect` with form values, but i want it to happen _onBlur_ and not _onChange_. With the new api to subscribe to form value changes for uncontrolled mode, I was wondering about adding a different kind of subscription- an _onBlur_ one.

Gut things to consider- depends on input type i suppose. For most use cases this should probably trigger onBlur for text inputs but t i think as a user the expected behavior is onChange for Switches, Checkboxes, and Radio buttons.

I'm open to discussion/ helping implement a feature like this one. 

Thanks!

**Comment by scottyeck** [comment]
Also curious to explore this. The way I've considered it, there should be an `onValuesChange` at the form-level that allows you to hook into the changes, and then the ability to push new uncontrolled values back to the form in some configurable way on a field-by-field basis. (For example, sometimes I may want a `TextInput` that publishes via `onBlur`, whereas other times it might be `onChange`.)

In any case, I think the original question is a good one. Curious to hear additional thoughts.
